utils.py

```diff
@@ -19,8 +19,13 @@ def filter_country_sort_year(df, country):
     """
     df = df.loc[df["country_name"] == country]
 
-    df = df[df.year >= START_YEAR]
+    df = df[df.year >= start_year]
+    if not df.empty:
+        earliest_year = df['year'].min()
```

Comment:
@weilu 
Although I do think it is nice to have the empty check here,
should we also fix the aggregate output data here?

https://github.com/dime-worldbank/mega-boost/blob/a2b9c7baa4c55be8877dc72cc29791674e48c03e/cross_country_aggregate_dlt.py#L114-L122

Here, we are joining the expenditure data and the poverty data. The earliest year is based on the expenditure data, but when joined, the 'earliest year' may not be present in the poverty data, causing some inconsistencies.

 Should we update the earliest year for the resulting aggregate data in boost repo?

Reply:
It's a right join where the expenditure data is on the right so all rows from the expenditure data will be retained even if some poverty data is missing for earlier years, so it should be fine. Am I missing something?

Reply:
Got it! I think I was confusing myself with the fix. I am going to approve shortly!